config/routes.rb

```diff
@@ -6,9 +6,9 @@
   # You can have the root of your site routed with "root"
   root 'static_pages#index'
   resources :posts do
-    resources :comments, :only => :create 
+    resources :comments, :only => [:create, :edit]
```

Comment:
I don't like this because this would pass through both the comment_id and the post_id.  This is redundant information and you'd want to add validations the URL is properly managed:

```
/posts/:post_id/comments/:id/edit    # we're passing through more info than we want, we don't need the post_id, we can get it from the comment.
```